src/Futhark/CLI/Eval.hs

```diff
@@ -0,0 +1,155 @@
+module Futhark.CLI.Eval (main) where
+  
+import Control.Exception
+import Control.Monad
+import Control.Monad.Except
+import Control.Monad.Free.Church
+
+import Data.Maybe
+import Data.Text qualified as T
+
+import Futhark.Compiler
+import Futhark.MonadFreshNames
+-- import Futhark.Util (toPOSIX)
+import Futhark.Util.Options
+
+import Language.Futhark
+import Language.Futhark.Interpreter qualified as I
+import Language.Futhark.Parser
+import Language.Futhark.Semantic qualified as T
+import Language.Futhark.TypeChecker qualified as T
+
+import System.FilePath
+
+import Futhark.Pipeline
+
+import System.IO
+import Prelude
+-- import Language.Futhark.Interpreter (interpretExp, ExtOp)
+
+
+import Futhark.Util.Pretty
+import Data.List (foldl')
+
+main :: String -> [String] -> IO ()
+-- main _ _ = putStrLn "Hello World"
+main = mainWithOptions interpreterConfig options "options... <exprs...>" run
+  where
+    run [] _ = Nothing
+    run exprs config = Just $ runExprs exprs config
+
+runExprs :: [String] -> InterpreterConfig -> IO ()
+runExprs exprs cfg = do
+  -- putStrLn "Running expressions"
+  let InterpreterConfig _ file = cfg
+  -- putStrLn "File has come up."
+  maybe_new_state <- newFutharkiState cfg file
+  -- putStrLn "New state made"
+  (src, env, ctx) <- case maybe_new_state of
+    Left _ -> error "Unable to load file."
```

Comment:
This should not be an `error` (which is used only for internal bugs), but `hPutStrLn stderr` or something similar.

Reply:
I've added an `exitWith` as well.